config/daemon/ipv6.md

```diff
@@ -10,31 +10,102 @@ Before you can use IPv6 in Docker containers or swarm services, you need to
+> - IPv6 is only supported on Docker daemons running on Linux hosts.
 
-1.  Edit `/etc/docker/daemon.json`, set the `ipv6` key to `true` and the `fixed-cidr-v6` key
-    to your IPv6 subnet. In this example we are setting it to `2001:db8:1::/64`.
+## Daemon configuration
 
-    ```json
-    {
-      "ipv6": true,
-      "fixed-cidr-v6": "2001:db8:1::/64"
-    }
-    ```
+To enable IPv6, you must edit the Docker daemon configuration file located at
+`/etc/docker/daemon.json`. Configure the following parameters:
 
-    Save the file.
+```json
+{
+  "experimental": true,
+  "ipv6": true,
+  "ip6tables": true,
+  ...
+}
+```
 
-2.  Reload the Docker configuration file.
+You can optionally also configure the `fixed-cidr-v6` key, if you want to
+assign an IPv6 subnet to the default bridge network:
 
-    ```console
-    $ systemctl reload docker
-    ```
+```diff
+{
+  "experimental": true,
+  "ipv6": true,
+  "ip6tables": true,
++ "fixed-cidr-v6": "2001:db8:1::/64",
+  ...
+}
+```
+
+After saving the configuration file, reload the Docker daemon for your
+changes to take effect:
+
+```console
+$ systemctl reload docker
+```
 
 You can now create networks with the `--ipv6` flag and assign containers IPv6
 addresses using the `--ip6` flag.
 
+## Dynamic IPv6 subnet allocation
+
+If you want dynamic IPv6 subnet allocation, you must explicitly configure the
+`default-address-pools` parameter to include:
+
+- The default address pool values
+- One or more custom IPv6 supernets
+
+> **Note**
+>
+> Be aware that the following known limitations exist:
+>
+> - Supernets can't have a size larger than 80. This is due to an integer
+>   overflow in the Docker daemon. See
+>   [moby/moby#42801](https://github.com/moby/moby/issues/42801)
+> - The difference between the supernet length and the pool size can't be
+>   larger than 24. Otherwise, the daemon consumes all available memory. See
+>   [moby/moby#40275](https://github.com/moby/moby/issues/40275)
+
+The default address pool configuration is:
+
+```json
+{
+  "default-address-pools": [
+    { "base": "172.17.0.0/16", "size": 16 },
+    { "base": "172.18.0.0/16", "size": 16 },
+    { "base": "172.19.0.0/16", "size": 16 },
+    { "base": "172.20.0.0/16", "size": 16 },
+    { "base": "172.24.0.0/14", "size": 16 },
+    { "base": "172.28.0.0/14", "size": 16 },
+    { "base": "172.28.0.0/16", "size": 20 }
+  ]
+}
+```
+
+The following example shows a valid configuration with the default values and
+an IPv6 supernet, with a prefix length of 64 and a size of 80.
+
+```json
+{
+  "default-address-pools": [
+    { "base": "172.17.0.0/16", "size": 16 },
+    { "base": "172.18.0.0/16", "size": 16 },
+    { "base": "172.19.0.0/16", "size": 16 },
+    { "base": "172.20.0.0/16", "size": 16 },
+    { "base": "172.24.0.0/14", "size": 16 },
+    { "base": "172.28.0.0/14", "size": 16 },
+    { "base": "172.28.0.0/16", "size": 20 },
+    { "base": "2001:db8::/64", "size": 80 }
+  ]
+}
+```
```

Comment:
This could also be clarified better?

The user will get IPv6 subnets of `/80`, the network prefix is always `/64` for ULA and GUA networks right? The last 16 bits of that are for for the subnet ID range, but could be reduced for longer routing prefix (_but that doesn't change the size of the network prefix_).

I don't have too much experience with IPv6, but I think what I've said is applicable to most of the audience that is inexperienced with IPv6 and typically only interested in GUA and ULA subnets?

AFAIK, when the subnet is less than `/64`, some routing specific features would be lost. Instead leaning on NDP proxy to route through the hosts `/64` (for GUA addresses at least) as the public gateway to a subnet "slice" (_I don't know the correct jargon here, technically you've been using supernet for `base` and subnet for `size`, but I've only seen IPv6 `/64` blocks referred to as subnets elsewhere_).

For ULA, you could assign IPv6 `/64` networks but I recall there's an open bug report about subnets with a `base` larger than `/64` being fixed (_instead of allowing multiple `size: 64` networks within that subnet ID range from a `/48`, you only got the 1?_). Perhaps something else that needs a note?

Reply:
I think given the likely audience that some additional awareness might need to be made for the example here.

The IPv6 subnet used as the `base` is reserved for documentation, but I don't expect inexperienced IPv6 readers to know this. In fact I've seen some third-party resources share this as a solution to get IPv6 addresses preferred over IPv4 for connections (_since for ULA networks, `/etc/gai.conf` can determine which is preferred_).

A small section or note could compliment this clarifying that it's an example subnet reserved for documentation, and briefly touch on ULA and GUA addresses?

It doesn't help that private network range subnets are shown prior to it, which while relevant may contribute to user confusion when the documentation IPv6 subnet is interleaved within that example? `fd00:d0ca::/64` could be used..at the risk of that spreading as a more "official" config before any equivalent IPv6 ULA subnet is considered for upstreaming :sweat_smile: